batch.go

```diff
@@ -0,0 +1,58 @@
+package schnorrkel
+
+import (
+	"errors"
+
+	"github.com/gtank/merlin"
+	r255 "github.com/gtank/ristretto255"
+)
+
+// VerifyBatch batch verifies the given signatures
+func VerifyBatch(transcripts []*merlin.Transcript, signatures []*Signature, pubkeys []*PublicKey) (bool, error) {
+	if len(transcripts) != len(signatures) || len(signatures) != len(pubkeys) || len(pubkeys) != len(transcripts) {
+		return false, errors.New("the number of transcripts, signatures, and public keys must be equal")
+	}
+
+	zero := r255.NewElement().Zero()
+
+	// compute H(R_i || P_i || m_i)
+	hs := make([]*r255.Scalar, len(transcripts))
```

Comment:
I believe even though you are creating a certain amount Scalar you will still have allocations when this is used. You can pre-seed it to avoid allocation by doing something like: 


```go
hs := make([]*r255.Scalar, len(transcripts))
scalar := make([]r255.Scalar, len(transcripts))
for i := range hs {
		hs[i] = &svals[i]
	}
```